src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj

```diff
@@ -8,6 +8,8 @@
   <ItemGroup>
     <PackageReference Include="Moq" Version="$(MoqVersion)" />
     <ProjectReference Include="..\src\Microsoft.Extensions.Primitives.csproj" />
+
+    <TrimmerRootDescriptor Include="$(ILDescriptorsPath)ILLink.Descriptors.Castle.xml" />
```

Comment:
Is there a way to detect if the test has a reference to `Moq`, and add this automatically if it does? That way this doesn't need to be copied everywhere.

Reply:
We could look for `Moq` in `@(PackageReference)`. I tried to find all the references to `Moq`, and `*Castle.xml`:

```
$ git grep -E "PackageReference.*Moq|Castle.xml"

src/libraries/Microsoft.Extensions.Configuration.FileExtensions/tests/Microsoft.Extensions.Configuration.FileExtensions.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/Microsoft.Extensions.Configuration.FileExtensions/tests/Microsoft.Extensions.Configuration.FileExtensions.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Configuration/tests/Microsoft.Extensions.Configuration.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/Microsoft.Extensions.Configuration/tests/Microsoft.Extensions.Configuration.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.DependencyModel/tests/Microsoft.Extensions.DependencyModel.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.FileProviders.Composite/tests/Microsoft.Extensions.FileProviders.Composite.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.FileProviders.Composite/tests/Microsoft.Extensions.FileProviders.Composite.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/Microsoft.Extensions.FileProviders.Physical/tests/Microsoft.Extensions.FileProviders.Physical.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Microsoft.Extensions.Hosting.Unit.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/Microsoft.Extensions.Http.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/Microsoft.Extensions.Http.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/Microsoft.Extensions.Logging/tests/Common/Microsoft.Extensions.Logging.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/Microsoft.Extensions.Logging/tests/Common/Microsoft.Extensions.Logging.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/Microsoft.Extensions.Primitives/tests/Microsoft.Extensions.Primitives.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj:    <TrimmerRootDescriptor Include="$(ILLinkDescriptorsPath)ILLink.Descriptors.Castle.xml" />
src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
src/libraries/System.Text.Encoding/tests/System.Text.Encoding.Tests.csproj:    <PackageReference Include="Moq" Version="$(MoqVersion)" />
```

.. and there seem to be some projects that either don't need the linker xml, or maybe I didn't get to failing test because of `Moq` yet.

Should we add that irrespective? or maybe doing it as-needed per project is better?